java-frontend/src/main/java/org/sonar/java/resolve/SecondPass.java

```diff
@@ -109,6 +109,12 @@ private void complete(JavaSymbol.TypeJavaSymbol symbol) {
 
     if ((symbol.flags() & Flags.INTERFACE) == 0) {
       symbol.members.enter(new JavaSymbol.VariableJavaSymbol(Flags.FINAL, "super", type.supertype, symbol));
+    } else {
+      // JLS9 - 15.12.1 : To support invocation of default methods in superinterfaces, the TypeName may also refer
+      // to a direct superinterface of the current class or interface, and the target is that superinterface.
+      symbol.members.enter(new JavaSymbol.VariableJavaSymbol(Flags.FINAL, "super", type, symbol));
+      // Note: The above "super" symbol will always be qualified when referenced. e.g. A.super.hashCode()
```

Comment:
mentioning the super-interfaces explicitly can also come from the fact that an interface/class can extend/implement multiple interfaces. Having it mentioned explicitly remove any ambiguity.